### PR TITLE
Full Accountless Flow

### PR DESCRIPTION
**Pull request by stevenelleman**

### CODE REVIEW

File: apps/frontend/src/pages/people/index.tsx [hunk 1]
```diff
@@ -17,17 +28,17 @@ const PeoplePage: React.FC = () => {
   useEffect(() => {
     const fetchConnections = async () => {
       const user = await storage.getUser();
-      if (!user) {
+      const unregisteredUser = await getUnregisteredUser();
+      if (!user && !unregisteredUser) {
         return;
```
Comment (RiverRuby) [inline apps/frontend/src/pages/people/index.tsx:33]: Should this router push somewhere else?

File: apps/frontend/src/pages/tap.tsx [hunk 2]
```diff
@@ -63,15 +63,6 @@ const TapPage: React.FC = () => {
               return;
             }
 
-            // If user is not logged in, direct them to sign in
```
Comment (RiverRuby) [inline apps/frontend/src/pages/tap.tsx:66]: If we're not doing location taps in accountless then why did we remove this?

File: apps/frontend/src/pages/tap.tsx [hunk 3]
```diff
@@ -114,8 +96,11 @@ const TapPage: React.FC = () => {
             // Save tap to local storage
             await storage.addUserTap(response);
```
Comment (RiverRuby) [inline apps/frontend/src/pages/tap.tsx:97]: shouldn't this not happen if user is not registered?

File: apps/frontend/src/pages/tap.tsx [hunk 4]
```diff
@@ -114,8 +96,11 @@ const TapPage: React.FC = () => {
...
-            // Update leaderboard entry
-            await updateTapLeaderboardEntry(response.chipIssuer);
+            if (user) {
+              // If no user, skip updating tap leaderboard.
+              // For the team being assume unregistered users won't be recorded
+              await updateTapLeaderboardEntry(response.chipIssuer);
+            }
 
             // Save tap to populate modal upon redirect
             await storage.saveTapInfo({ tapParams, tapResponse: response });
```
Comment (RiverRuby) [inline apps/frontend/src/pages/tap.tsx:106]: same with this?

File: apps/frontend/src/pages/people/index.tsx (hunk 1 above)
Comment (RiverRuby) [inline apps/frontend/src/pages/people/index.tsx:33]: shouldn't we boot them off this page in this case?

File: apps/frontend/src/pages/people/[username].tsx [hunk 5]
```diff
@@ -206,34 +206,57 @@ const UserProfilePage: React.FC = () => {
...
+            savedTapInfo.tapResponse.userTap?.ownerUsername === username
+          ) {
+            logClientEvent("user-profile-tap-chip-modal-shown", {});
+            setTapInfo(savedTapInfo);
+            setShowCommentModal(true);
+          }
+        } else if (unregisteredUser) {
+          // In unregistered user case, cannot do PSIs which require user object
+          setConnection(unregisteredUser.connections[username]);
+          const savedTapInfo = await storage.loadSavedTapInfo();
```
Comment (RiverRuby) [inline apps/frontend/src/pages/people/[username].tsx:248]: I see, the earlier saveTpa it's used to transfer over info to this page, makes sense

File: apps/frontend/src/pages/tap.tsx (hunk 4 above)
Comment (RiverRuby) [reply apps/frontend/src/pages/tap.tsx:106]: i see this is used to get info to the people page

**Review by RiverRuby — COMMENTED** [review]
I'm feeling a bit confused on the tap.tsx flow, seems a bit unprotected especially re: locationTaps. Which will exist in the museum so we should make sure that doesn't get user into a weird state

File: apps/frontend/src/lib/auth/register.ts [hunk 6]
```diff
@@ -141,3 +166,77 @@ export async function registerUser({
...
+
+  const user: UnregisteredUser = {
+    signaturePrivateKey,
+    encryptionPrivateKey,
+    lastMessageFetchedAt: new Date(),
+    userData: {
+      signaturePublicKey,
+      encryptionPublicKey,
+    },
+    connections: {},
```
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/frontend/src/lib/auth/register.ts:184]: I'll make an empty record for locations.

File: apps/frontend/src/pages/tap.tsx (hunk 2 above)
Comment (stevenelleman) [reply apps/frontend/src/pages/tap.tsx:66]: Adding back in.

File: apps/frontend/src/lib/auth/register.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/frontend/src/lib/auth/register.ts:184]: Good catch.

File: apps/frontend/src/pages/people/index.tsx (hunk 1 above)
Comment (stevenelleman) [reply apps/frontend/src/pages/people/index.tsx:33]: Good catch.

File: apps/frontend/src/lib/storage/types/user/index.ts [hunk 7]
```diff
@@ -22,6 +26,21 @@ export const UserSchema = z.object({
 
 export type User = z.infer<typeof UserSchema>;
 
+export const UnregisteredUserSchema = z.object({
+  signaturePrivateKey: z.string(),
+  encryptionPrivateKey: z.string(),
+  lastMessageFetchedAt: z.coerce.date(),
+  userData: UnregisteredUserDataSchema,
+  connections: z.record(z.string(), ConnectionSchema),
+  locations: nullToUndefined(z.record(z.string(), LocationSchema)),
```
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/frontend/src/lib/storage/types/user/index.ts:38]: out of scope.

File: apps/frontend/src/lib/auth/register.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/frontend/src/lib/auth/register.ts:184]: added, addressed, ty.

File: apps/frontend/src/pages/tap.tsx (hunk 2 above)
Comment (stevenelleman) [reply apps/frontend/src/pages/tap.tsx:198]: added back, tested and it works, addressed

File: apps/frontend/src/pages/tap.tsx (hunk 3 above)
Comment (stevenelleman) [reply apps/frontend/src/pages/tap.tsx]: we need this for adding the connection to the unregistered user

File: apps/frontend/src/pages/people/index.tsx (hunk 1 above)
Comment (stevenelleman) [reply apps/frontend/src/pages/people/index.tsx:37]: I now boot, addressed

**Review by RiverRuby — APPROVED** [review]
LGTM

**Pull request merged**